tokio/tests/rt_unstable_metrics.rs

```diff
@@ -295,37 +295,34 @@ fn worker_noop_count() {
 }
 
 #[test]
-#[ignore] // this test is flaky, see https://github.com/tokio-rs/tokio/issues/6470
 fn worker_steal_count() {
     // This metric only applies to the multi-threaded runtime.
-    //
-    // We use a blocking channel to backup one worker thread.
-    use std::sync::mpsc::channel;
 
     let rt = threaded_no_lifo();
     let metrics = rt.metrics();
 
-    rt.block_on(async {
-        let (tx, rx) = channel();
-
-        // Move to the runtime.
-        tokio::spawn(async move {
-            // Spawn the task that sends to the channel
-            //
-            // Since the lifo slot is disabled, this task is stealable.
-            tokio::spawn(async move {
-                tx.send(()).unwrap();
-            });
+    let mut successfully_spawned_stealable_task = false;
 
-            // Blocking receive on the channel.
-            rx.recv().unwrap();
-        })
-        .await
-        .unwrap();
+    rt.block_on(async {
+        for _ in 0..10 {
+            // The call to `try_spawn_stealable_task` may time out, which means
+            // that the sending task couldn't be scheduled due to a deadlock in
+            // the runtime.
+            // This is expected behaviour, we just retry until we succeed or
+            // exhaust all tries, the latter causing this test to fail.
+            if try_spawn_stealable_task().await.is_ok() {
+                successfully_spawned_stealable_task = true;
+                break;
+            }
+        }
     });
 
     drop(rt);
 
+    if !successfully_spawned_stealable_task {
+        panic!("exhausted every try to schedule the stealable task");
+    }
+
     let n: u64 = (0..metrics.num_workers())
         .map(|i| metrics.worker_steal_count(i))
         .sum();
```

Comment:
We should recreate the runtime on failure. Otherwise a failed round could result in this number being wrong.

Reply:
Ah, I guess this clould be the case for #6916 too.

Reply:
> Ah, I guess this clould be the case for https://github.com/tokio-rs/tokio/pull/6916 too.

I don't _think_ the `global_queue_depth` in #6916 could be tainted by a task from the previous try to block the runtime. That would require both tasks from the next try to be scheduled before the one unscheduled (but already queued) task of the previous try, which I don't believe can happen? If that could happen, the test would be racy. I'm not exactly sure what guarantees Tokio has for task execution order. Even if my assumption is true, if we are relying on an implementation detail I think we should recreate the runtime in that test, too, to bullet proof it against future changes.

Reply:
If the deadlock scenario is what I mentioned [here](https://github.com/tokio-rs/tokio/pull/6876#issuecomment-2408993124), then one injection queue task will remain in the queue. So I think starting another iteration without refreshing the runtime could taint the queue.

Reply:
> So I think starting another iteration without refreshing the runtime could taint the queue.

That would require both tasks from the current try to block the runtime to be executed before the one remaining task from the previous try. Is that possible under the consideration that a task blocks the worker it is executed on? 

(Sorry for my ignorance of the implementation details, I've yet to fully grasp the execution model of the multi-threaded runtime)

Reply:
Yes, but it's probably pretty unlikely.

Reply:
Okay, then I should definitely revisit #6916 and recreate the runtime on each try. Should I make a new PR or do the changes here?

Reply:
Up to you. I would probably make a new PR.

Reply:
#6936 for reference.